File: js/videos.js
// videos.js
const { dumpSchema } = require("./utils/index.js");
const BraveAPI = require("./API.js");

// Create a new instance of the BraveAPI class
const brave = new BraveAPI({
    subscription_token: process.env.BRAVE_KEY
});

// Setup the search parameters
const params = {
    q: "function calling with groq",
    freshness: "pm",
};

// Perform the search
brave.videos(params).then(results => {
    // Log the results
    console.log(dumpSchema(results));
});

File: python/videos.py
# videos.py
import os
from API import BraveAPI
from utils.main import dump_schema

# Create an instance of the BraveAPI class
brave = BraveAPI({
    'subscription_token': os.environ['BRAVE_KEY']
})

# Setting up parameters
params = {
    'q': "function calling with groq",
    'freshness': "py"
}

# Performing the videos search
results = brave.videos(params)

# Dumping and printing the schema of the results
print(dump_schema(results))
